deployment/stacks/api.ts

```diff
@@ -0,0 +1,423 @@
+      "Access-Control-Allow-Origin": "'*'",
+      "Access-Control-Allow-Headers": "'*'",
+    },
+  });
+
+  api.addGatewayResponse("Default5XXResponse", {
+    type: apigateway.ResponseType.DEFAULT_5XX,
+    responseHeaders: {
+      "Access-Control-Allow-Origin": "'*'",
+      "Access-Control-Allow-Headers": "'*'",
+    },
+  });
+
+  const environment = {
+    NODE_OPTIONS: "--enable-source-maps",
+    BOOTSTRAP_BROKER_STRING_TLS: brokerString,
+    stage,
+    TEMPLATE_BUCKET: templateBucket ? templateBucket.bucketName : "",
+    WP_FORM_BUCKET: wpFormBucket.bucketName,
+    SAR_FORM_BUCKET: sarFormBucket.bucketName,
+    ...Object.fromEntries(
+      tables.map((table) => [`${table.id}Table`, table.name])
+    ),
+  };
+
+  const additionalPolicies = [
+    new iam.PolicyStatement({
+      effect: iam.Effect.ALLOW,
+      actions: [
+        "dynamodb:DescribeTable",
+        "dynamodb:Query",
+        "dynamodb:Scan",
+        "dynamodb:GetItem",
+        "dynamodb:PutItem",
+        "dynamodb:UpdateItem",
+        "dynamodb:DeleteItem",
+      ],
+      resources: [
+        ...tables.map((table) => table.arn),
+        ...tables.map((table) => `${table.arn}/index/*`),
+      ],
+    }),
+    new iam.PolicyStatement({
+      effect: iam.Effect.ALLOW,
+      actions: ["s3:GetObject", "s3:ListBucket", "s3:PutObject"],
+      resources: [
+        ...(templateBucket
+          ? [
+              templateBucket.bucketArn,
+              `${templateBucket.bucketArn}/templates/*`,
+            ]
+          : []),
+        ...[
+          `${wpFormBucket.bucketArn}/formTemplates/*`,
+          wpFormBucket.bucketArn,
+          `${wpFormBucket.bucketArn}/formTemplates/*`,
+          `${wpFormBucket.bucketArn}/fieldData/*`,
+          sarFormBucket.bucketArn,
+          `${sarFormBucket.bucketArn}/formTemplates/*`,
+          `${sarFormBucket.bucketArn}/fieldData/*`,
+        ],
+      ],
+    }),
+    new iam.PolicyStatement({
+      effect: iam.Effect.ALLOW,
+      actions: [
+        "dynamodb:DescribeStream",
+        "dynamodb:GetRecords",
+        "dynamodb:GetShardIterator",
+        "dynamodb:ListShards",
+        "dynamodb:ListStreams",
+      ],
+      resources: tables.map((table) => table.streamArn).filter(isDefined),
+    }),
+    new iam.PolicyStatement({
+      effect: iam.Effect.ALLOW,
+      actions: ["s3:GetObject", "s3:PutObject", "s3:ListBucket"],
+      resources: [
+        ...(templateBucket ? [templateBucket.bucketArn] : []),
+        ...[
+          wpFormBucket.bucketArn,
+          sarFormBucket.bucketArn,
+          `${wpFormBucket.bucketArn}/fieldData/*`,
+          `${sarFormBucket.bucketArn}/fieldData/*`,
+        ],
+      ],
+    }),
+  ];
+
+  const commonProps = {
+    brokerString,
+    stackName: `${service}-${stage}`,
+    api,
+    environment,
+    additionalPolicies,
+    iamPermissionsBoundary,
+    iamPath,
+  };
+
+  const requestValidator = new apigateway.RequestValidator(scope, `Validator`, {
```

Comment:
We haven't done this for other api gateways. Do we have to? If so we'll need to add to SEDS and HCBS.

Reply:
MFP had this before, I was trying to match.